crates/rustc_codegen_c/src/helper.h

```diff
@@ -0,0 +1,7 @@
+/* Some helper macros for the generated code */
+
+/** cast from unsigned to signed
+  * example: `__rust_utos(uint32_t, int32_t, x, INT32_MAX)`
+  */
+#define __rust_utos(u, s, v, m) \
+    ((v) <= (m) ? ((s)v) : ((s)((u)(v) - (u)(m) - 1)))
```

Comment:
Could you expand the docs with a bit about what this is doing? (I know you have it in the rust doc comments, but it would be good to have it next to the source).